c2corg_ui/templates/waypoint/edit.html

```diff
@@ -780,7 +780,14 @@ <h2 class="heading show-phone"><span translate>summary</span></span></h2>
         </section>
         
         <p class="action-buttons float-buttons">
-          <button type="submit" class="btn orange-btn btn-lg float-button" ng-disabled="editForm.$invalid || editForm.$pristine" tooltip-placement="left" uib-tooltip="{{'Save' | translate}}">
+          <button type="submit" class="btn orange-btn btn-lg float-button" 
+            ng-if="!editForm.$invalid && (waypoint.locales[0].title && waypoint.waypoint_type && waypoint.locales[0].lang && waypoint.elevation && waypoint.lonlat.latitude && waypoint.lonlat.longitude)" tooltip-placement="left" uib-tooltip="{{'Save' | translate}}">
+            <span class="glyphicon glyphicon-ok"></span>
+          </button>
+          <button type="button" class="btn btn-lg orange-btn float-button" 
+                  ng-style="(editForm.$invalid || (!waypoint.locales[0].title || !waypoint.waypoint_type || !waypoint.locales[0].lang || !waypoint.elevation || !waypoint.lonlat.longitude || !waypoint.lonlat.latitude)) && {'opacity': 0.5}" 
+                  tooltip-placement="left" uib-tooltip="{{'What is missing?' | translate}}" 
+                  ng-if="editForm.$invalid || (!waypoint.locales[0].title || !waypoint.waypoint_type || !waypoint.locales[0].lang || !waypoint.elevation || !waypoint.lonlat.longitude || !waypoint.lonlat.latitude)" ng-click="editCtrl.checkMissingProps(waypoint)">
```

Comment:
Can't you move this check if all required fields are given into a method? And can't you use `checkMissingProps` for that purpose?